### PR TITLE
Fix annotation translation when the value is 0

### PR DESCRIPTION
**Pull request by pauzinl**
This patch fixes translation for annotations, when all values of annotation
are 0.

### CODE REVIEW

**Comment by pauzinl** [comment]
@MrSidims @KornevNikita take a look please

File: lib/SPIRV/SPIRVWriter.cpp
Comment (KornevNikita) [inline lib/SPIRV/SPIRVWriter.cpp:2584]: Typo

File: test/transcoding/annotate_attribute.ll
Comment (KornevNikita) [inline test/transcoding/annotate_attribute.ll:53]: Could you also add a check if the value is `false` (`[[clang::annotate("my_annotation", false)]]`). And add a case for a mix with 0 e.g. `{ i32 0, i32 1, i32 0 }` to make sure that the issue occures only if all values are zero?

File: lib/SPIRV/SPIRVWriter.cpp
Comment (pauzinl) [reply lib/SPIRV/SPIRVWriter.cpp]: Thanks! Fixed it

**Review by MrSidims — APPROVED** [review]
LGTM
Will merge after @KornevNikita approval

**Review by KornevNikita — APPROVED** [review]
Thanks! Also LGTM!

**Pull request merged**
